hazelcast/src/test/java/com/hazelcast/partition/strategy/AttributePartitioningStrategyIntegrationTest.java

```diff
@@ -142,6 +144,32 @@ public void testJsonObject() {
         assertEquals(0, nonOwners.get(1).getMap("test").getLocalMapStats().getOwnedEntryCount());
     }
 
+    @Test
+    public void testDynamicPartitioningAttributesConfiguration() {
+        final List<PartitioningAttributeConfig> attributeConfigs =
+                singletonList(new PartitioningAttributeConfig("name"));
+        final MapConfig mapConfig = new MapConfig("test2").setPartitioningAttributeConfigs(attributeConfigs);
+
+        client().getConfig().addMapConfig(mapConfig);
+
+        for (final HazelcastInstance instance : instances()) {
+            assertTrueEventually(() -> assertEquals(mapConfig, instance.getConfig().getMapConfig("test2")));
```

Comment:
if I understand `AddDynamicConfigOperation` implementation correctly, this test can be flaky if invocation of `getMapConfig("test2")` happens before `addMapConfig` is replicated. In such case `test2` could have default config. It there anything that prevents it?

Reply:
When `addMapConfig(mapConfig);` returns, map config to be present in the cluster. Why did you think otherwise looking at the `AddDynamicConfigOperation`?

Reply:
I assumed that `broadcastConfigAsync` is fire-and-forget and overlooked that it waits for completion. So the test should be fine. Thanks for the clarification.